src/Wave.cpp

```diff
@@ -0,0 +1,147 @@
+#include "Wave.hpp"
+#include "Utilities.hpp"
+
+using namespace std;
+using namespace gazebo;
+using namespace gazebo_usv;
+using namespace ignition::math;
+
+Wave::Wave(Wave::EffectParameters const parameters) : mParameters(parameters) {}
+
+Wave::~Wave()
+{
+    if (mWaveVelocitySubscriber)
+    {
+        mWaveVelocitySubscriber->Unsubscribe();
+    }
+    if (mWaveFrequencySubscriber)
+    {
+        mWaveFrequencySubscriber->Unsubscribe();
+    }
+}
+
+void Wave::load(ModelPtr const _model, transport::NodePtr const _node, sdf::ElementPtr const _sdf)
+{
+    mModel = _model;
+    mNode = _node;
+    mLink = getReferenceLink(mModel, _sdf);
+    
+    auto pluginName = _sdf->Get<std::string>("name");
+    string topicName = std::regex_replace(pluginName, std::regex("__"), "/") ;
+    
+    string topicNameVelocity = topicName + "/wave_velocity";
+
+
+    if (mWaveVelocitySubscriber)
+    {
+        mWaveVelocitySubscriber->Unsubscribe();
+    }
+    mWaveVelocitySubscriber = mNode->Subscribe("/" + topicNameVelocity, &Wave::readWaveVelocity, this);
+    
+    string topicNameFrequency = topicName + "/wave_frequency";
+    if (mWaveFrequencySubscriber)
+    {
+        mWaveFrequencySubscriber->Unsubscribe();
+    }
+    mWaveFrequencySubscriber = mNode->Subscribe("/" + topicNameFrequency, &Wave::readWaveFrequency, this);
+    
+    auto worldName = mModel->GetWorld()->Name();
+    gzmsg << "Wave: receiving wave commands from /gazebo/"
+          << worldName << "/" << topicNameVelocity << endl;
+    gzmsg << "Wave: receiving wave commands from /gazebo/"
+          << worldName << "/" << topicNameFrequency << endl;
```

Comment:
The logs are confusing, maybe do a single Wave: receiving wave commands from /gazebo/ with both of the topic names.